posts/2022-09-27-instant-on-220011-beta.adoc

```diff
@@ -0,0 +1,1171 @@
+
+For this example, we will be using an official image from the IBM Container Registry (ICR), `icr.io/appcafe/open-liberty:beta-instanton-java17-openj9-ubi`, as the parent image. This image is tagged with the word `beta`, meaning it includes all the Liberty beta features as well as all the Liberty features from the `full` image.  The image is tagged with the word `instanton`, meaning it includes all the prerequisites for producing a checkpoint process image, for example, the necessary `criu` binaries.
+
+The `Dockerfile` for the getting-started application already exists at `finish/Dockerfile`. Edit the existing `finish/Dockerfile` and change the `FROM` instruction to use the parent image `icr.io/appcafe/open-liberty:beta-instanton-java17-openj9-ubi`. After saving the `Dockerfile` it should look something like this:
+
+.Dockerfile
+[source]
+----
+FROM icr.io/appcafe/open-liberty:beta-instanton-java17-openj9-ubi
+
+ARG VERSION=1.0
+ARG REVISION=SNAPSHOT
+
+LABEL \
+  org.opencontainers.image.authors="Your Name" \
+  org.opencontainers.image.vendor="IBM" \
+  org.opencontainers.image.url="local" \
+  org.opencontainers.image.source="https://github.com/OpenLiberty/guide-getting-started" \
+  org.opencontainers.image.version="$VERSION" \
+  org.opencontainers.image.revision="$REVISION" \
+  vendor="Open Liberty" \
+  name="system" \
+  version="$VERSION-$REVISION" \
+  summary="The system microservice from the Getting Started guide" \
+  description="This image contains the system microservice running with the Open Liberty runtime."
+
+COPY --chown=1001:0 src/main/liberty/config/ /config/
+COPY --chown=1001:0 target/*.war /config/apps/
+
+RUN configure.sh
+
+----
+
+[#app-image]
+=== Building the application container image
+Before building the image, an understanding of what is required to perform a checkpoint and restore is necessary. For `criu` to be able to do its job of taking a checkpoint of a process and restoring a process, the `criu` binary must be granted additional link:https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux_atomic_host/7/html/container_security_guide/linux_capabilities_and_seccomp[Linux capabilities]. In particular, for Open Liberty, it needs to be granted `cap_checkpoint_restore`, `cap_net_admin` and `cap_sys_ptrace`. The Open Liberty InstantOn beta image includes the `criu` binary with the necessary capabilities granted to the `criu` binary file. In order for the `criu` binary to be given access its assigned capabilities at runtime, the container which is running `criu` must also be granted the necessary capabilities when it is launched. This can be done in one of two ways:
+
+. Use a privileged container using the `--privileged` option
```

Comment:
I'll move this down to the section `Create the application checkpoint image`.  The information is not necessary for the section on creating the initial application image (without the checkpoint).

Reply:
Actually, I need to introduce this here because I wanted to establish that all the proceeding `podman` commands must be run with `sudo` or as root.  This includes the initial `podman` commands for building the application image.  So I'm going to leave this here for now.